Add Zig to the wheel

jedisct1 opened the pull request:
Zig missing from the wheel was a bit unexpected, so here's a wheel part for it.

jedisct1 committed:
Add Zig to the wheel

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/boyanio/wasm-wheel?pullRequest=25) <br/>All committers have signed the CLA.

jedisct1 committed:
Zig: use wasm32-freestanding, no need for WASI

boyanio commented on src/langs/zig/Dockerfile:
```diff
@@ -0,0 +1,7 @@
+FROM kassany/bookworm-ziglang:latest AS wheel-part-zig
```
Can we set it to a specific version instead of latest? What I experience with other languages is that in newer versions there are sometimes changes that break the wasm compilation

jedisct1 replied:
Done.

boyanio commented on src/langs/zig/wheel-part.zig:
```diff
@@ -0,0 +1,9 @@
+extern fn jsrandom() callconv(.C) f64;
```
I use this approach for some of the other languages as well (because their native way for generating random numbers does not compile to wasm). I am not familiar with Zig, but I was wondering if there was a native way to generate random numbers in Zig that we could use here

jedisct1 replied:
Done!

jedisct1 committed:
Pin a version for the Zig compiler, use built-in RNG

Pin the Zig toolchain to 0.11.0

Use JavaScript only to get the seed, but then use the Zig standard
library to generate numbers out of it.
